lesson19/src/js/slider.js

```diff
@@ -0,0 +1,210 @@
+  slideImageArea.appendChild(sliderImage);
+}
+
+function changeActivePagination(currentIndex) {
+  const paginationCurrent = document.getElementById("js-current");
+  paginationCurrent.textContent = currentIndex;
+
+}
+
+function changeActiveImage(target) {
+  const active = document.querySelector(".is-active");
+  active.classList.remove("is-active");
+  const changeSlide = document.querySelector(`[data-slide-index="${target}"]`);
+  changeSlide.classList.add("is-active");
+}
+
+function changeActivePaginationBullet(target){
+  const paginationBullet = document.querySelector('[aria-current="true"');
+  paginationBullet.setAttribute('aria-current', false);
+  const pagiNations = document.querySelector(`[data-pagination-index="${target}"]`);
+  pagiNations.setAttribute("aria-current", true);
+
+}
+
+function clickPaginationBullet(){
+  const paginationBullets = document.getElementById('js-pagination');
+
+  paginationBullets.addEventListener('click', (e) => {
+    const targetElement = e.target.classList.contains('paginationbullet__bar');
+
+    if(targetElement){
+      const activePaginationBullet = document.querySelector('[aria-current="true"]');
+      const clickedPaginationBulletIndex = e.target.getAttribute('data-pagination-index');
+      activePaginationBullet.setAttribute('aria-current','false');
+      e.target.setAttribute('aria-current','true');
+      changeActiveSlider(clickedPaginationBulletIndex);
+      changeActivePagination(clickedPaginationBulletIndex);
+      resetAutoPlaySlide();
+    }
+  },true);
+};
+
+function arrowButtonIsDisabled(index) {
+  const slideImages = [...document.getElementsByClassName("slider__item")];
+  const firstIndex = 1;
+  const lastIndex = slideImages.length;
+  const prevButton = document.getElementById("js-button_previous");
+  const nextButton = document.getElementById("js-button_next");
+
+  prevButton.disabled = Number(index) === firstIndex;
+  nextButton.disabled = Number(index) === lastIndex;
+}
+
+function renderSlidContents(slideImage){
+  renderSlideImage(slideImage);
+  renderPagination(slideImage.length);
+  renderPaginationBullet(slideImage.length);
+  renderSlideButton();
+}
+
+function addSlideAction(){
+  addChangeButtonEvent();
+  clickPaginationBullet();
+  autoPlayslide();
+}
+
+function addChangeButtonEvent() {
+  const arrowButtons = document.querySelector(".slider-button");
+
+  arrowButtons.addEventListener("click", (event) => {
+    const activeImage = document.querySelector(".is-active");
+    let currentIndex = Number(activeImage.dataset.slideIndex);
+    currentIndex =
+      event.target.id === "js-button_next" ? ++currentIndex : --currentIndex;
+    changeActivePagination(currentIndex);
+    changeActiveSlider(currentIndex);
+    resetAutoPlaySlide()
+  });
+}
+
+function changeActiveSlider(slideTarget) {
+  arrowButtonIsDisabled(slideTarget);
+  changeActiveImage(slideTarget);
+  changeActivePaginationBullet(slideTarget);
+}
+
+let autoPlayID;
```

Comment:
IMO
I feel that it is best to avoid defining a let outside of a function. In such cases I group them into objects.
```js
const autoPlay = {
 intervalId: null,
}
```
When to use
```js
function autoPlayslide() {
  autoPlay.intervalId = setInterval(() => {
    const slideImages = document.querySelectorAll(".slider__item").length;
    const activeImage = document.querySelector(".is-active");
    let currentIndex = Number(activeImage.dataset.slideIndex);
    currentIndex = currentIndex === slideImages ? 1 : currentIndex + 1;
    changeActiveSlider(currentIndex);
    changeActivePagination(currentIndex);
  }, 3000);
}
```

Reply:
I couldn't think of a way to make let an object and declare it with const. I will study a lot!
This has been fixed.

[1aa8b47](https://github.com/nyattsu72/hanson-frontend-cource/pull/27/commits/1aa8b47e599bbaaf285f8391b4ab5a6488a361ff)